tests/raster/test_geotiff.py

```diff
@@ -0,0 +1,307 @@
+        np.testing.assert_equal(ts2.data[band_name], data[:, 10:15, 12:17])
+
+    def test_read_decoded_image_stack(self):
+        """
+        Tests reading and decoding of an image stack.
+
+        """
+        num_files = 25
+        xsize = 60
+        ysize = 50
+        band_names = ['data1', 'data2']
+
+        ds_tile = Tile(ysize, xsize, SpatialRef(4326), name='0')
+        ds_mosaic = MosaicGeometry.from_tile_list([ds_tile])
+
+        dims = ['time', 'y', 'x']
+        coords = {'time': pd.date_range('2000-01-01', periods=num_files),
+                  'y': ds_tile.y_coords,
+                  'x': ds_tile.x_coords}
+        data = np.ones((num_files, ysize, xsize))
+        attr1 = {'unit': 'dB', 'scale_factor': 2, 'add_offset': 3, 'fill_value': -9999}
+        attr2 = {'unit': 'dB', 'fill_value': -9999}
+        ds = xr.Dataset({band_names[0]: (dims, data, attr1), band_names[1]: (dims, data, attr2)}, coords=coords)
+
+        with GeoTiffWriter(ds_mosaic, data=ds, stack_dimension='time', dirpath=self.path,
+                           fn_pattern='{time}.tif', fn_formatter={'time': lambda x: x.strftime('%Y%m%d')}) as gt_writer:
+            gt_writer.export()
+            filepaths = list(gt_writer.file_register['filepath'])
+
+        with GeoTiffReader.from_filepaths(filepaths) as gt_reader:
+            ts1 = gt_reader.select_px_window(0, 0, width=10, height=10, inplace=False)
+            ts1.read(bands=(1, 2), band_names=band_names, auto_decode=True)
+            ts2 = gt_reader.select_px_window(10, 12, width=5, height=5, inplace=False)
+            ts2.read(bands=(1, 2), band_names=band_names, auto_decode=True)
+            np.testing.assert_equal(ts1.data[band_names[1]], data[:, :10, :10])
+            np.testing.assert_equal(ts2.data[band_names[1]], data[:, 10:15, 12:17])
+
+        with GeoTiffReader.from_filepaths(filepaths) as gt_reader:
+            ts1 = gt_reader.select_px_window(0, 0, width=10, height=10, inplace=False)
+            ts1.read(bands=(1, 2), band_names=band_names, auto_decode=False)
+            ts2 = gt_reader.select_px_window(10, 12, width=5, height=5, inplace=False)
+            ts2.read(bands=(1, 2), band_names=band_names, auto_decode=False)
+            np.testing.assert_equal(ts1.data[band_names[0]], data[:, :10, :10])
+            np.testing.assert_equal(ts2.data[band_names[0]], data[:, 10:15, 12:17])
+
+        data = data * 2. + 3.
+        with GeoTiffReader.from_filepaths(filepaths) as gt_reader:
+            ts1 = gt_reader.select_px_window(0, 0, width=10, height=10, inplace=False)
+            ts1.read(bands=(1, 2), band_names=band_names, auto_decode=True)
+            ts2 = gt_reader.select_px_window(10, 12, width=5, height=5, inplace=False)
+            ts2.read(bands=(1, 2), band_names=band_names, auto_decode=True)
+            np.testing.assert_equal(ts1.data[band_names[0]], data[:, :10, :10])
+            np.testing.assert_equal(ts2.data[band_names[0]], data[:, 10:15, 12:17])
```

Comment:
should be a separate or parameterized test

Reply:
Done.